src/CodeGen_Hexagon.cpp

```diff
@@ -2025,5 +2060,25 @@ void CodeGen_Hexagon::visit(const NE *op) {
     }
 }
 
+void CodeGen_Hexagon::visit(const Allocate *op) {
+    if (op->memory_type == MemoryType::VTCM && !op->new_expr.defined()) {
```

Comment:
Rather than check the existing new_expr is not defined, should we instead assert that it is not inside the body of this if? Silently failing to allocate VTCM seems like a bad behavior.

Reply:
I am not sure if I understand this correctly.
new_expression not defined check is there to avoid infinite recursion because I'm calling new_alloc.accept(this) at the end of the if block (which will also enter the if block and so on). This check ensures I visit the if block only once per VTCM allocate node.

Reply:
Hmm, I see the problem, but it would be good to find a way to add an assert rather than silently fail. Maybe just add a TODO for now.

Reply:
We discussed this yesterday, not sure if you heard it but we decided that it's OK to simply skip the VTCM logic if there's already a new_expr, which is responsible for allocating correctly.

Reply:
Thanks Dillon. Pranav conveyed the same to me.